pinot-core/src/test/java/org/apache/pinot/core/transport/QueryRoutingTest.java

```diff
@@ -286,6 +293,89 @@ public void testServerDown()
     assertEquals(_serverRoutingStatsManager.fetchNumInFlightRequestsForServer(serverId).intValue(), 0);
   }
 
+  @Test
+  public void testSkipUnavailableServer()
+      throws IOException, InterruptedException {
+    // Using a different port is a hack to avoid resource conflict with other tests, ideally queryServer.shutdown()
+    // should ensure there is no possibility of resource conflict.
+    int port = 12346;
```

Comment:
I don't love this workaround, but using the same port as other tests impacts test isolation. It looks like `queryServer.shutDown()` doesn't properly shutdown and block. 

I'm open to suggestions, as I am pretty unfamiliar w/ netty and did not find a clean solution

Reply:
Other tests in the same class are sharing the same port, so I guess there are something else impacting this particular test method